### DISCUSSION TITLE
React componentWillUnmount() analog?

### DISCUSSION THREAD

**Discussion by spokvulcan** [DioxusLabs/dioxus#340] [Q&A] [answered]
Maybe a stupid question)

How to perform some logic when component will be unmounted?

**Comment by scottbot95** [comment, marked as answer]
It's kinda hacky and there probably should be a `use_umount`/`use_drop`, but you can implement `Drop` on the value you send to [ScopeState.use_hook](https://docs.rs/dioxus/0.2.4/dioxus/prelude/struct.ScopeState.html#method.use_hook) to get a callback when the component is dropped.

I actually ended up writing my own hook to do this in a somewhat convenient way:
```rust
pub struct UseDrop {
    callback: Box<dyn Fn()>,
}

impl Drop for UseDrop {
    fn drop(&mut self) {
        (self.callback)();
    }
}

pub fn use_drop(cx: &ScopeState, callback: impl Fn() + 'static) {
    cx.use_hook(|_| UseDrop {
        callback: Box::new(callback),
    });
}
```
